tools/predict_net.py

```diff
@@ -0,0 +1,207 @@
+from time import time
+
+import numpy as np
+import pandas as pd
+import cv2
+import torch
+
+import slowfast.utils.checkpoint as cu
+import slowfast.utils.distributed as du
+import slowfast.utils.logging as logging
+import slowfast.utils.misc as misc
+from slowfast.datasets import loader
+from slowfast.models import model_builder
+# from slowfast.utils.meters import TestMeter
+
+logger = logging.get_logger(__name__)
+
+def imresize(im, dsize):
+    '''
+    Resize the image to the specified square sizes and 
+    maintain the original aspect ratio using padding.
+
+    Args:
+
+        im -- input image.
+        dsize -- output sizes, can be an integer or a tuple.
+
+    Returns:
+
+        resized image.
+    '''
+    if type(dsize) is int:
+        dsize = (dsize, dsize)
+        
+    im_h, im_w, _ = im.shape
+    to_w, to_h = dsize
+    scale_ratio = min(to_w/im_w, to_h/im_h)
+    new_im = cv2.resize(im,(0, 0), 
+                        fx=scale_ratio, fy=scale_ratio, 
+                        interpolation=cv2.INTER_AREA)
+    new_h, new_w, _ = new_im.shape
+    padded_im = np.full((to_h, to_w, 3), 128)
+    x1 = (to_w-new_w)//2
+    x2 = x1 + new_w
+    y1 = (to_h-new_h)//2
+    y2 = y1 + new_h
+    padded_im[y1:y2, x1:x2, :] = new_im 
+    
+    return padded_im
+
+class VideoReader(object):
+
+    def __init__(self, source):
+        self.source = source
+        try:  # OpenCV needs int to read from webcam
+            self.source = int(source)
+        except ValueError:
+            pass
+
+    def __iter__(self):
+        self.cap = cv2.VideoCapture(self.source)
+        # self.cap.set(cv2.CAP_PROP_FRAME_WIDTH, 1280)
+        # self.cap.set(cv2.CAP_PROP_FRAME_HEIGHT, 720)
```

Comment:
Remove the comments if it is not used. Maybe adding an option to the config if you prefer.